python/cudf_polars/cudf_polars/experimental/sort.py

```diff
@@ -284,19 +314,23 @@ def insert_partition(
 
         by = options["by"]
 
+        stream = get_joined_cuda_stream(upstreams=(df.stream, sort_boundaries.stream))
+
         splits = find_sort_splits(
             df.select(by).table,
             sort_boundaries.table,
             partition_id,
             options["order"],
             options["null_order"],
+            stream=stream,
         )
         packed_inputs = split_and_pack(
             df.table,
             splits=splits,
             br=context.br,
-            stream=DEFAULT_STREAM,
+            stream=stream,
         )
+        # TODO: figure out handoff with rapidsmpf
```

Comment:
A bit more context here: we need to look into the pylibcudf methods called by rapidsmpf and ensure that they all accept a `stream` argument, and that we have a way to pass that through.